SwEnt-Group8/Street-work-app
# Feat/basic version of progression mvvm

**Summary**
This Pull request introduces the first step for the gamification of the app. It introduces a ViewModel for a new class called a "Progression". This MVVM will be used to reward the user with medals, points and achievements when specific actions are done in the app.

**Features**
- A first ViewModel for Progressions linked to every users
- Basic database operation (create and fetch the progression of a user)
- A first function to reward the user with medals (called checkScore)

**How to test**
The Ui linked to this viewModel is not ready yet, but you can use the functions in the main and see in the database the result. You should be able to create and fetch a "Progression" Object. The checkScore function can be used to add achievements to a User.
